File: backendworker/NHSOnline.Backend.Worker/GpSystems/Suppliers/Vision/Models/Appointments/References.cs
using System.Collections.Generic;
using System.Xml.Serialization;

namespace NHSOnline.Backend.Worker.GpSystems.Suppliers.Vision.Models.Appointments
{
    public class References
    {
        [XmlElement(ElementName = "owner", Namespace = "urn:vision")]
        public List<Owner> Owners { get; set; }

        [XmlElement(ElementName = "session", Namespace = "urn:vision")]
        public List<SlotSession> Sessions { get; set; }

        [XmlElement(ElementName = "location", Namespace = "urn:vision")]
        public List<Location> Locations { get; set; }

        [XmlElement(ElementName = "slotType", Namespace = "urn:vision")]
        public List<SlotType> SlotTypes { get; set; }
    }
}

File: bddtests/src/main/kotlin/mocking/vision/models/appointments/References.kt
package mocking.vision.models.appointments

import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(namespace = "urn:vision")
data class References(
        @XmlElement(namespace = "urn:vision")
        var location: List<Location>? = null,
        @XmlElement(namespace = "urn:vision")
        var owner: List<Owner>? = null,
        @XmlElement(namespace = "urn:vision")
        var session: List<Session>? = null,
        @XmlElement(namespace = "urn:vision")
        var slotType: List<SlotType>? = null
)


